libderections-common/src/main/java/com/mapbox/navigation/route/common/NavigationRoute.kt

```diff
@@ -0,0 +1,782 @@
+package com.mapbox.navigation.route.common
+
+import android.content.Context
+import androidx.annotation.FloatRange
+import androidx.annotation.IntRange
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.MapboxDirections
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.core.exceptions.ServicesException
+import com.mapbox.core.utils.TextUtils
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.route.common.extension.getUnitTypeForLocale
+import com.mapbox.navigation.utils.extensions.inferDeviceLocale
+import okhttp3.EventListener
```

Comment:
can we not leak okhttp and retfrofit dependencies through public apis?

Reply:
it won't be leaked, because for user we provide separate callbacks